daemon/response.go

```diff
@@ -365,13 +369,49 @@ func SnapNotFound(snapName string, err error) Response {
 // operation is requested for which no revivision can be found
 // in the given context (e.g. request an install from a stable
 // channel when this channel is empty).
-func SnapRevisionNotAvailable(snapName string, err error) Response {
+func SnapRevisionNotAvailable(snapName string, rnaErr *store.RevisionNotAvailableError) Response {
+	var value interface{} = snapName
+	kind := errorKindSnapRevisionNotAvailable
+	msg := rnaErr.Error()
+	if len(rnaErr.Releases) != 0 && rnaErr.Channel != "" {
+		thisArch := arch.UbuntuArchitecture()
+		values := map[string]interface{}{
+			"snap-name":    snapName,
+			"action":       rnaErr.Action,
+			"channel":      rnaErr.Channel,
+			"architecture": thisArch,
+		}
+		archOK := false
+		releases := make([]map[string]interface{}, 0, len(rnaErr.Releases))
+		for _, c := range rnaErr.Releases {
+			if c.Architecture == thisArch {
+				archOK = true
+			}
+			releases = append(releases, map[string]interface{}{
+				"architecture": c.Architecture,
+				"channel":      c.Name,
+			})
+		}
+		// we return all available releases (arch x channel)
+		// as reported in the store error, but we hint with
+		// the error kind whether there was anything at all
+		// available for this architecture
```

Comment:
It's curious that we have logic to split a common case into two different errors here, and then on the client side we pick up the two different errors and have extra logic to merge them back into a single case. This seems to indicate to me that we'd benefit from keeping the error as a common case in the first place (preserving the single "no revision" kind), and then just decorate the error with some more metadata to make it easier to construct the error.

For example, we might have a simple field like:
```
"alternatives": ["channel", "arch"],
```

This is also nice because we can in fact have a situation in which _both_ a different channel or a different architecture would solve the problem. We cannot present that well via a single error kind.